assets/src/edit-story/components/panels/stylePreset/panel.js

```diff
@@ -0,0 +1,203 @@
+        };
+      }
+      if (
+        addedPresets.fillColors?.length > 0 ||
+        addedPresets.textColors?.length > 0 ||
+        addedPresets.textStyles?.length > 0
+      ) {
+        updateStory({
+          properties: {
+            stylePresets: {
+              textStyles: [...textStyles, ...addedPresets.textStyles],
+              fillColors: [...fillColors, ...addedPresets.fillColors],
+              textColors: [...textColors, ...addedPresets.textColors],
+            },
+          },
+        });
+      }
+    },
+    [
+      fillColors,
+      textStyles,
+      textColors,
+      isText,
+      selectedElements,
+      updateStory,
+      stylePresets,
+    ]
+  );
+
+  const handleApplyPreset = useCallback(
+    (preset) => {
+      if (isText) {
+        // @todo Determine this in a better way.
+        // Only style presets have background text mode set.
+        const isStylePreset = preset.backgroundTextMode !== undefined;
+        updateElementsById({
+          elementIds: selectedElementIds,
+          properties: isStylePreset ? { ...preset } : { color: preset },
+        });
+      } else {
+        updateElementsById({
+          elementIds: selectedElementIds,
+          properties: { backgroundColor: preset },
+        });
+      }
+    },
+    [isText, selectedElementIds, updateElementsById]
+  );
+
+  const colorPresets = isText ? textColors : fillColors;
+  const hasColorPresets = colorPresets.length > 0;
+  const hasPresets = hasColorPresets || textStyles.length > 0;
+
+  useEffect(() => {
+    // If there are no colors left, exit edit mode.
+    if (isEditMode && !hasPresets) {
+      setIsEditMode(false);
+    }
+  }, [hasPresets, isEditMode]);
+
+  // @todo This is temporary until the presets haven't been implemented fully with multi-selection.
+  if (!isText && !isShape && selectedElements.length > 1) {
+    return null;
+  }
+
+  const getEventHandlers = (preset) => {
+    return {
+      onClick() {
+        handlePresetClick(preset);
+      },
+      onKeyDown(evt) {
+        if (evt.keyCode === 'Enter' || evt.keyCode === 'Space') {
+          handlePresetClick(preset);
+        }
+      },
+    };
+  };
+
+  const handlePresetClick = (preset) => {
+    if (isEditMode) {
+      handleDeletePreset(preset);
+    } else {
+      handleApplyPreset(preset);
+    }
+  };
+
+  // @Todo confirm initial height.
+  return (
+    <Panel
+      name="stylepreset"
+      initialHeight={Math.min(200, window.innerHeight / 3)}
+      resizeable
+    >
+      <PresetsHeader
+        handleAddColorPreset={handleAddColorPreset}
+        stylePresets={stylePresets}
+        isEditMode={isEditMode}
+        setIsEditMode={setIsEditMode}
+      />
+      <Presets
+        isEditMode={isEditMode}
+        stylePresets={stylePresets}
+        getEventHandlers={getEventHandlers}
+        isText={isText}
+        textContent={isText ? selectedElements[0].content : ''}
```

Comment:
Element's `content` is not text, but HTML. Is this handled upstream? IMHO we should either extract plaintext here or rename and support HTML upstream. I think there's already `stripHTML` or such if you decide to just get plaintext here.

Reply:
Done!